community/kernel/src/main/java/org/neo4j/kernel/impl/api/cursor/TxNodeCursor.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.api.cursor;
+
+import org.neo4j.graphdb.Direction;
+import org.neo4j.kernel.api.cursor.LabelCursor;
+import org.neo4j.kernel.api.cursor.NodeCursor;
+import org.neo4j.kernel.api.cursor.NodeValue;
+import org.neo4j.kernel.api.cursor.RelationshipCursor;
+import org.neo4j.kernel.impl.api.KernelStatement;
+
+/**
+ * Overlays transaction state on a {@link NodeCursor}.
+ */
+public class TxNodeCursor
+        implements NodeCursor
+{
+    private final KernelStatement statement;
+    private NodeCursor nodeCursor;
+
+    private final NodeValueImpl value = new NodeValueImpl();
+
+    public TxNodeCursor( KernelStatement statement )
+    {
+        this.statement = statement;
+    }
+
+    public TxNodeCursor init(NodeCursor nodeCursor)
+    {
+        this.nodeCursor = nodeCursor;
+        return this;
+    }
+
+    @Override
+    public boolean next()
+    {
+        return nodeCursor.next();
+    }
+
+    @Override
+    public void close()
+    {
+        nodeCursor.close();
+        statement.releaseNodeCursor(this);
+    }
+
+    @Override
+    public NodeValue get()
```

Comment:
This interface might be misleading since the returned `NodeValue` is not stable but changes everytime you call `next()`. isn't it perhaps better having `getId()`, `labels`, etc on the  `TxNodeCursor` directly?